Jaden taking over for Alisha - Feature: Add village dropdown filter on Bidding Homepage (Done Jaden)

alishawalunj opened the pull request:
(PRIORITY High) Jaden taking over for Alisha - Feature: Add village dropdown filter on Bidding Homepage (Done Jaden)

URL: /lbdashboard/bidding

## Related PRs (if any):
This frontend PR is not related to any backend PR.

## Main changes explained:
- Village dropdown filter in the LBDashboard header on the bidding page - selecting a village navigates to that village's detail page
- Village details page showing the village map image, amenities list, and properties grid
- "Go back to Map" link on the village details page redirects to the masterplan
- Double-clicking a village card on the masterplan page navigates to that village's detail page
- Redux wiring: action, reducer, constants added for fetching village data from the backend API
- Merged with latest development branch - resolved conflicts in Header.jsx, reducers/index.js, routes.jsx

## How to test:
1. Check out current branch
2. Run `npm install` and `npm start`
3. Clear site data/cache
4. Log in as admin user
5. Navigate to `/lbdashboard/bidding`
6. Verify "Filter by Village" dropdown is visible in the header - select a village and confirm it redirects to that village's detail page
7. On the village detail page, verify the village map image, amenities, and properties are displayed
8. Click "Go back to Map" - confirm it redirects to `/lbdashboard/masterplan`
9. On the masterplan page, single-click a village card to select it, then click it again to confirm it redirects to that village's detail page

## Note:
Frontend implementation. Village data is served from the backend API - the dropdown falls back to fixed village list if the API is unavailable.

## Screenshots or videos of changes:
https://github.com/user-attachments/assets/30ba63b1-eec5-4d27-b14a-b38e78054d1a

<img width="1115" height="859" alt="img2" src="https://github.com/user-attachments/assets/bd69430c-39db-4663-a41a-9a31ea194d5d" />
<img width="1401" height="874" alt="img3" src="https://github.com/user-attachments/assets/bd25767d-9ab4-4ddb-a9a2-54dd57944762" />
<img width="1679" height="896" alt="img4" src="https://github.com/user-attachments/assets/56e78d41-106e-414a-8807-8e2ba8f4fe40" />
<img width="1671" height="905" alt="img5" src="https://github.com/user-attachments/assets/a1f06928-a3d8-4c99-91c6-8bad196ed1ce" />

alishawalunj committed:
Add village dropdown filter

Prem203 commented:
Tested the PR with an admin account and everything works as expected. Great job!

<img width="1899" height="870" alt="image" src="https://github.com/user-attachments/assets/1e3e9013-b007-4dab-b2f5-e73e845d7e7f" />

<img width="1871" height="858" alt="image" src="https://github.com/user-attachments/assets/09d921fb-7285-4f35-911a-2aaf4c5d248d" />

<img width="1850" height="859" alt="image" src="https://github.com/user-attachments/assets/905d0dc2-dafe-41bb-9254-e2c4d3b21f97" />

ChiragBellara commented:
Reviewed PR #4124 

1. The page loads successfully and Filter by Village dropdown is visible and functions as expected.
<img width="720" height="450" alt="initial page load and filter works" src="https://github.com/user-attachments/assets/200e4a22-5222-4d79-ad78-d69d3055ab92" />

2. The village page shows all the details and images as expected.
<img width="1440" height="900" alt="village screen shows all the details" src="https://github.com/user-attachments/assets/b20c26cd-6fa4-4150-9e73-c0d9101bd264" />

3. The Go To Map redirect works and does redirect to the Master Plan page. On double clicking the village on the Master 
Plan page, I do get redirected to the specific village page.
<img width="1440" height="900" alt="Master Plan page works as expected" src="https://github.com/user-attachments/assets/fe0e464d-071e-45f7-94dd-30a1bfdbb4b7" />

aryanrachala54 reviewed (dismissed):
Everything works well, including the dropdown, village details page, and navigation. All functions are as expected.

<img width="1920" height="958" alt="Screenshot 2025-09-27 141330" src="https://github.com/user-attachments/assets/b76e89f0-7e6f-4d40-a794-31a72705d989" />
<img width="1920" height="955" alt="Screenshot 2025-09-27 141408" src="https://github.com/user-attachments/assets/5880dedc-c5f0-47f2-8caf-b00017c41fde" />
<img width="1920" height="958" alt="Screenshot 2025-09-27 141521" src="https://github.com/user-attachments/assets/6e7660d4-fbde-4e42-a5df-28d20c48bc8d" />

sircarmart reviewed (dismissed):
All filter options working as intended and showing details + image on their pages. Double clicking any of the villages in the master plan redirects to the expected page. Everything looks good.


https://github.com/user-attachments/assets/cb03fd03-f96a-44b9-97ef-14f01ee1ef93

Nhujarski reviewed (approved):
I tested the PR using the above steps. Filter by village is there, it functions as expected as well as the link and double clicking a village. The routes are working correctly as well. Great job!

<img width="1440" height="771" alt="filter by village" src="https://github.com/user-attachments/assets/90a673
<img width="1433" height="751" alt="village screen" src="https://github.com/user-attachments/assets/19bf2172-2c2a-42bf-8f4d-ff3b16073ca4" />
bf-498a-436e-babb-f8209b13aae6" />
<img width="1434" height="752" alt="master plan" src="https://github.com/user-attachments/assets/a151ed2d-eee0-4389-b606-1ce43967dd52" />
<img width="1427" height="753" alt="shipping village " src="https://github.com/user-attachments/assets/86c060ac-3ad8-4d78-b6d8-25d0d0fc48a0" />

abdel-lall reviewed (dismissed):
LGTM!
<img width="2905" height="1654" alt="Screenshot 2025-09-27 215357" src="https://github.com/user-attachments/assets/681906e8-c23c-4b6f-b048-d9f056465f61" />

Aswin20010 reviewed (dismissed):
Reviewed PR #4124 (Village Dropdown Filter and Details Page Feature). Verified addition of the new dropdown filter for villages on the Bidding page, proper routing to the Village Details page, and display of village images, amenities, and properties. Confirmed “Go Back to Map” navigation works and double-click on village image cards redirects correctly from the MasterPlan page. Checked data fetching, reducer integration, and CSS consistency. No UI or logic regressions observed — approved the PR.

<img width="1420" height="858" alt="Screenshot 2025-10-21 at 12 34 01 PM" src="https://github.com/user-attachments/assets/a02cf6d6-1ba0-4987-b89a-1efb71a866cb" />
<img width="1414" height="861" alt="Screenshot 2025-10-21 at 12 35 10 PM" src="https://github.com/user-attachments/assets/eb4d2a62-ff3e-40dd-8707-877e06f22fbe" />
<img width="1388" height="861" alt="Screenshot 2025-10-21 at 12 35 26 PM" src="https://github.com/user-attachments/assets/fa2bb151-2bf9-492d-9f44-ad27f2f63440" />

alishawalunj committed:
fix: update root reducer to include village dropdown state

alishawalunj committed:
Fix VillageDetails and DropdownFilter layout regression after CSS Modules migration

Anusha-Gali reviewed (changes requested):
Hi Alisha,

I have reviewed your PR locally and there were few issues that i would like to point out:
- The landing page does not appear to be the same as yours.
- The listing page is not found
- The carousel images have different sizes rather than a consistent size
- In dark mode the main page in similar fashion as the page would in light mode
- The dropdown filter is not adapted to dark mode
- Similarly the map page is not adapted to dark mode as well as all the respective village pages.
I have attached here a video depicting all issue. Do have a look.

https://github.com/user-attachments/assets/03ecf98f-ecb9-4afa-8dfe-79e6724de5bd

rohanrastogi311 reviewed (changes requested):
Hi Alisha, 

Unfortunately no villages are appearing on this page. 

<img width="1727" height="1039" alt="Image" src="https://github.com/user-attachments/assets/41e12ba6-e3e4-460a-9571-952097ea1774" />

alishawalunj committed:
fix: resolved sonarqube issues

Jaden300 committed:
merge development into village-dropdown-filter and fix quality gate issues

- resolve conflicts in Header.jsx, reducers/index.js, routes.jsx, lock files
- take development Header.jsx (already has village dropdown via FIXED_VILLAGES)
- add villageDetailsReducer to index.js under lbdashboard reducers
- add VillageDetails import and route to routes.jsx
- fix: dispatch error.message string instead of raw Error object in villageDetailsAction.js
- fix: remove unused React import and rename name variable to villageName in DropdownFilter.jsx

HemanthNidamanuru reviewed (changes requested):
Hi Jaden, 


I tested this PR locally and found the following issues:

The landing page looks entirely different from what is described, with no villages displayed.
The "Filter by Village" dropdown is not visible in the header.
Pages are not properly aligned in dark mode.

<img width="1917" height="1037" alt="Image" src="https://github.com/user-attachments/assets/5b4587b3-5fde-4d32-b77c-b75afd7a342e" />

<img width="1918" height="1031" alt="Image" src="https://github.com/user-attachments/assets/f18d3bdf-efe3-4b86-a4e3-a904027e1626" />

DeepighaJ reviewed (changes requested):
Tested the PR locally and on selecting a filter it doesn't navigate to other page and so not able to proceed with testing the other functionality mentioned.
<img width="1600" height="852" alt="Image" src="https://github.com/user-attachments/assets/6f0eddf3-50ba-46f9-a97c-065a705ea643" />
<img width="1600" height="852" alt="Image" src="https://github.com/user-attachments/assets/d6b6c48c-2ac1-4617-aa8c-bba7f8811464" />

Jaden300 committed:
fix village dropdown navigation, dark mode, and empty villages fallback

- replace Header select+Go button with VillageDropdownFilter so selecting a village auto-navigates without needing to click Go
- add FIXED_VILLAGES fallback to VillageDropdownFilter so dropdown is never empty when backend is unavailable
- add dark mode to DropdownFilter select (transparent bg, white text, blue border)
- fix BiddingHomepage dark mode - was using plain string 'dark-mode' so CSS variables never activated, now uses styles['dark-mode']
- add dark mode to VillageDetails page (dark bg, light text on container/info/property cards)

HemanthNidamanuru reviewed (changes requested):
Hi Jaden, 

I tested this PR locally. The 'Filter by Village' dropdown, village detail page (map, amenities, properties), 'Go back to Map' redirect, and village card selection/redirect are all working as expected. However, the masterplan page is not aligned in dark mode. 


<img width="1911" height="1040" alt="Image" src="https://github.com/user-attachments/assets/397ec34f-6c77-443f-90f6-299e569f394c" />

<img width="1918" height="1030" alt="Image" src="https://github.com/user-attachments/assets/c09b4539-d412-4f84-a2c6-b94af3c0c087" />

<img width="1908" height="1037" alt="Image" src="https://github.com/user-attachments/assets/262f03e5-f586-4b82-b017-b889c933686d" />

<img width="1918" height="1091" alt="Image" src="https://github.com/user-attachments/assets/e4601c56-2524-4c2c-8c2f-6b5ce87f24ca" />

<img width="1918" height="1042" alt="Image" src="https://github.com/user-attachments/assets/8a0e1ce4-b3d2-4c77-a2d8-6113538c61b1" />

<img width="1918" height="1032" alt="Image" src="https://github.com/user-attachments/assets/0bacfc9c-3f22-4b46-9d3d-567d4bd7903d" />

<img width="1906" height="1031" alt="Image" src="https://github.com/user-attachments/assets/5715d0a8-9b02-41a9-bc08-2ad1925cc935" />

<img width="1902" height="1041" alt="Image" src="https://github.com/user-attachments/assets/d47fb409-69f0-48f3-bd10-7bfecbb1c815" />

<img width="1907" height="1037" alt="Image" src="https://github.com/user-attachments/assets/cbd5121b-1fd1-428d-99e0-25d160e775a3" />

DeepighaJ reviewed (changes requested):
Tested the PR locally. 
On selecting the village filter, detailed page opens and displays the details.
On clicking back to map it navigates to /lbdashboard/masterplan and on single click on card displays the village detail which is **needs contrast improvement in dark mode**. On second click it displays the details as expected.
<img width="1600" height="852" alt="Image" src="https://github.com/user-attachments/assets/bac7995b-9145-478b-9f31-0a62433e3687" />
<img width="1600" height="852" alt="Image" src="https://github.com/user-attachments/assets/a6988bc8-3712-4447-9949-168da7771bf1" />
<img width="1600" height="852" alt="Image" src="https://github.com/user-attachments/assets/eca2dfc3-0f1c-4491-8d7c-1c1d0cf51ec0" />
<img width="1600" height="852" alt="Image" src="https://github.com/user-attachments/assets/f7710dd3-692f-4b55-ab40-8c6dd208c188" />

Jaden300 committed:
Fix MasterPlan dark mode alignment and selected village card contrast

HemanthNidamanuru reviewed (approved):
Hi Jaden,

I retested this PR locally.

The previously reported dark mode alignment issue in masterplan page has been resolved, and everything is working fine now.


<img width="1906" height="1041" alt="Image" src="https://github.com/user-attachments/assets/7b003f31-d1ae-40ee-8d26-f513b9136f98" />

<img width="1908" height="1031" alt="Image" src="https://github.com/user-attachments/assets/842ea8a6-187c-45e9-9899-9c987c5f17a8" />

<img width="1918" height="1040" alt="Image" src="https://github.com/user-attachments/assets/62cf0b4b-830e-4e80-a81e-49e4ffa14291" />

Jaden300 committed:
fix(merge): resolve conflicts with development branch
